lightweight/calendareventsmodel/calendareventsmodel.cpp

```diff
@@ -64,9 +64,26 @@ NemoCalendarEventsModel::NemoCalendarEventsModel(QObject *parent) :
     connect(&mUpdateDelayTimer, SIGNAL(timeout()), this, SLOT(update()));
 
     QString privilegedDataDir = QString("%1/.local/share/system/privileged/Calendar/mkcal/db").arg(QDir::homePath());
-    mWatcher->addPath(privilegedDataDir);
+    if (QFile::exists(privilegedDataDir)) {
+        if (!mWatcher->addPath(privilegedDataDir)) {
+            qWarning() << "CalendarEventsModel: error adding filesystem watcher for calendar db";
+        }
+    } else {
+        qWarning() << "CalendarEventsModel not following database changes, dir not found:" << privilegedDataDir;
+    }
+
     QSettings settings("nemo", "nemo-qml-plugin-calendar");
-    mWatcher->addPath(settings.fileName());
+
+    if (!QFile::exists(settings.fileName())) {
+        // forcefully create a settings file so changes can be followed
```

Comment:
Would it be better to just touch the file instead of writing dummy values to it?